File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    com = dict()
    
    for name in completion:
        if name not in com:
            com[name] = 1
        else:
            com[name] += 1
    
    for part in participant:
        if part not in com or com[part] == 0:
            answer = part
            break
        
        com[part] -= 1
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> participants = new HashMap<>();
        
        for (String name : participant) {
            int value = participants.getOrDefault(name, 0);
            participants.put(name, value+1);
        }
        
        for (String name : completion) {
            int value = participants.get(name);
            if (value == 1) participants.remove(name);
            else participants.put(name, value-1);
        }
        
        for (Map.Entry<String, Integer> entry : participants.entrySet()) {
            answer =  entry.getKey();
        }
        
        return answer;
    }
}